src/pudl/transform/gridpathratoolkit.py

```diff
@@ -0,0 +1,156 @@
+"""Transformations of the GridPath RA Toolkit renewable generation profiles.
+
+Wind and solar profiles are extracted separately, but concatenated into a single table
+in this module, as they have exactly the same structure. The aggregation tables are also
+concatenated together.
+"""
+
+import pandas as pd
+from dagster import AssetCheckResult, AssetIn, asset, asset_check
+
+import pudl
+
+
+def _transform_capacity_factors(
+    capacity_factors: pd.DataFrame, utc_offset: pd.Timedelta
+) -> pd.DataFrame:
+    """Basic transformations that can be applied to many profiles.
+
+    - Construct a datetime column and adjust it to be in UTC.
+    - Reshape the table from wide to tidy format.
+    - Name columns appropriately.
+    """
+    # Convert timestamp to UTC before stacking!
+    # TODO: Ensure that we're using the same hour starting vs. ending convention
+    # here as we use in the FERC-714 and EPA CEMS data. Should this be reflected in
+    # the column names here and elsewhere, whatever the convention is?
+    datetime_cols = ["year", "month", "day", "hour"]
+    capacity_factors["datetime_utc"] = (
+        pd.DatetimeIndex(pd.to_datetime(capacity_factors.loc[:, datetime_cols]))
+        - utc_offset
+    )
+    capacity_factors = (
+        capacity_factors.drop(columns=datetime_cols)
+        .set_index("datetime_utc")
+        .stack()
+        .reset_index()
+    )
+    capacity_factors.columns = ["datetime_utc", "aggregation_key", "capacity_factor"]
+    capacity_factors = capacity_factors.astype({"aggregation_key": "string"})
+    return capacity_factors
+
+
+@asset(io_manager_key="pudl_io_manager")
+def core_gridpathratoolkit__hourly_aggregated_extended_capacity_factors(
```

Comment:
This table name is a mouthful, and without the specificity of the wind / solar / thermal technology it feels awfully vague. Are there better options?

If we think about how we'd want to name the tables in the event that we've got all of them in or we're doing the aggregations and extensions ourselves, what would they look like?  Would the aggregation and extension be part of the "output" stage? At what point do the wind and solar profiles come together? Is that only after aggregation?

Is it wrong that the table name is plural? I think I might have carried that over from "profiles" accidentally.

Other options?

* `core_gridpathratoolkit__hourly_solar_capacity_factor`
* `out_gridpathratoolkit__hourly_weather_dependent_capacity_factors` (wind, solar, thermal derates)